robot/packages/cloud_package_manager.go

```diff
@@ -43,7 +43,8 @@ type managedPackage struct {
 	modtime    time.Time
 }
 
-var placeholderRegexp = regexp.MustCompile(`^\$\{(packages(\.(ml_models|modules))?\.[\w_\-]+)\}`)
+// group 1: entire placeholder 2: type (if it exists + including the period) 3: the package name.
+var placeholderRegexp = regexp.MustCompile(`\$\{(packages(.ml_models|.modules)?\.([\w-/]+))\}`)
```

Comment:
the reason I didn't add the optional . before the type is because it makes ${packages..testmod} pass as a valid [placeholder](https://regex101.com/r/LkEuYn/1). Not sure what is better here, to do the trim on the type or check that there is no .. in the placeholder after we validate it? I kind of think trimming is cleaner

Reply:
I actually just realized that while taking a snack break, and came back to mention it, but you beat me to it!

```suggestion
var placeholderRegexp = regexp.MustCompile(`\$\{(packages(\.(ml_models|modules))?\.([\w/-]+))\}`)
```
Then then just update the `matches[?]` indexes below?

Reply:
thank you! yeah that works :) will fix and then do the changes below as well